### ISSUE TITLE
flycheck-compile: chktex did not succeed

### ISSUE THREAD

**Issue by liwoxa** [flycheck/flycheck#1841]
Thank you for taking the time to report an issue and improve Flycheck.  This template is for **actual bugs you observed**.  If you have trouble setting up Flycheck, or if you have a question, please use the relevant issue template instead.

## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description
Upon opening or saving a `tex` file, the following message pops:

```
Suspicious state from syntax checker tex-chktex: Flycheck checker tex-chktex returned 1, but its output contained no errors: ERROR: collate_byname<char>::collate_byname failed to construct for 

Sorry, but /Users/liwoxa/Plumbing/MiKTeX/bin/chktex did not succeed.

Try installing a more recent version of tex-chktex, and please open a bug report if the issue persists in the latest release.  Thanks!
```

The `flycheck-verify-setup` looks ok to me. The error seems to come down to `flycheck-compile`:

```
-*- mode: compilation; default-directory: "~/Desktop/Repositories/concurrent-optimisations/notes/interference/" -*-
Compilation started at Thu Dec  3 10:05:18

/Users/liwoxa/Plumbing/MiKTeX/bin/chktex --verbosity\=0 --quiet --inputfiles < /Users/liwoxa/Desktop/Repositories/concurrent-optimisations/notes/interference/texchk.tex
ERROR: collate_byname<char>::collate_byname failed to construct for 

Sorry, but /Users/liwoxa/Plumbing/MiKTeX/bin/chktex did not succeed.

Compilation exited abnormally with code 1 at Thu Dec  3 10:05:18
```

## Steps to reproduce
1. Open a `tex` file.
2. See error

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## System configuration
```
Syntax checkers for buffer texchk.tex in latex-mode:

First checker to run:

  tex-chktex
    - may enable:         yes
    - executable:         Found at /Users/liwoxa/Plumbing/MiKTeX/bin/chktex
    - configuration file: Not found

Checkers that could run if selected:

  tex-lacheck  select
    - may enable: yes
    - executable: Found at /Users/liwoxa/Plumbing/MiKTeX/bin/lacheck

Checkers that are compatible with this mode, but will not run until properly configured:

  textlint (automatically disabled) reset
    - may enable:         no
    - executable:         Not found
    - configuration file: Not found
    - textlint plugin:    @textlint/text

Flycheck Mode is enabled. Use SPC u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 32snapshot (package: 20201105.423)
Emacs version:    26.3
System:           x86_64-apple-darwin19.0.0
Window system:    ns
```

Emacs configuration:
- [ ] Plain Emacs / Custom configuration
- [x] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes
I am working in a non-admin user, with MikTeX installed locally. I made sure that `chktex` is updated, and also ran it from the terminal on a file and it worked fine:

```sh
$ chktex texchk.tex
ChkTeX v1.7.6 - Copyright 1995-96 Jens T. Berger Thielemann.
Compiled with POSIX extended regex support.
No errors printed; No warnings printed; No user suppressed warnings; No line suppressed warnings.
```

where `texchk.tex` is:

```latex
\documentclass{article}

\begin{document}
\(D(X)\)
\end{document}
```

[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by cpitclaudel** [comment]
Can you try running it from the command line with input on stdin?

```
chktex --verbosity=0 --quiet --inputfiles < texchk.tex
```

Does that work?

**Comment by liwoxa** [comment]
Doesn't produce any output. I guess that means it works?

**Comment by cpitclaudel** [comment]
Sounds like it.  Which is odd, because when flycheck runs it from Emacs you get the error message.  Can you run it from a shell within emacs (M-x shell)? If that works too, can you confirm that the full invocation also works within Emacs?

```
/Users/liwoxa/Plumbing/MiKTeX/bin/chktex --verbosity\=0 --quiet --inputfiles < /Users/liwoxa/Desktop/Repositories/concurrent-optimisations/notes/interference/texchk.tex
```

**Comment by liwoxa** [comment]
Indeed, I got:

```sh
ERROR: collate_byname<char>::collate_byname failed to construct for 

Sorry, but /Users/liwoxa/Plumbing/MiKTeX/bin/chktex did not succeed.
```

**Comment by liwoxa** [comment]
I'll mention that my emacs is installed through brew (so it is located in `usr/local/Cellar/emacs-plus@27/27.1/Emacs.app`) and I'm working in a non-admin account on my machine. Perhaps that's relevant.

**Comment by cpitclaudel** [comment]
> I got:

OK, then this is unrelated to Emacs: it happens in the terminal as well.  I would report the issue to the chktex people.

**Comment by liwoxa** [comment]
When I run
```
/Users/liwoxa/Plumbing/MiKTeX/bin/chktex --verbosity\=0 --quiet --inputfiles < /Users/liwoxa/Desktop/Repositories/concurrent-optimisations/notes/interference/texchk.tex
```
in the terminal window I don't get an error though.

**Comment by cpitclaudel** [comment]
Ah, so how did you run the command in Emacs?

**Comment by liwoxa** [comment]
As you asked, using `M-x shell` (actually `[SPC] [SPC] shell` because I'm using spacemacs).

Images of outputs attached for comparison:

Emacs shell:
<img width="831" alt="Screen Shot 2020-12-07 at 11 10 21" src="https://user-images.githubusercontent.com/48007333/101331534-d1afed80-387c-11eb-8cfd-ac4b220093bd.png">

Terminal:
<img width="584" alt="Screen Shot 2020-12-07 at 11 07 45" src="https://user-images.githubusercontent.com/48007333/101331581-de344600-387c-11eb-9c08-f04e5dfb4bc6.png">

**Comment by cpitclaudel** [comment]
Thanks.  I'm not familiar with this error (I don't use chktex), so I would suggest asking its developers.  It's likely due to a difference in environment variables: are you on macOS? (this type of errors tends to pop up on macOS because of the difference between terminal and graphical Emacs).

You could try comparing the output of `env` in the terminal and in M-x shell.

**Comment by liwoxa** [comment]
Indeed the output of `env` is different in all sorts of ways. Looked this issue up a bit and found [this reddit comment](https://www.reddit.com/r/emacs/comments/ixtk78/emacs_doesnt_inherit_shell_environment/g69c1m1/?utm_source=reddit&utm_medium=web2x&context=3). So I tried to open Emacs through the Terminal command line:
```
open /usr/local/Cellar/emacs-plus@27/27.1/Emacs.app
```
Lo and behold! It works!



Later on, I might try out the [exec-path-from-shell package](https://github.com/purcell/exec-path-from-shell) to see if I can avoid running Emacs from the command line.

Thank you for all the help, much appreciated!

**Comment by cpitclaudel** [comment]
Thanks for your patience!
